sr_port/hd.mpt

```diff
@@ -11,15 +14,16 @@
 %HD	;GT.M %HD utility - hexadecimal to decimal conversion program
 	;invoke at INT with %HD in hexadecimal to return %HD in decimal
 	;invoke at FUNC as an extrinsic function
+	;Note: Ignores any prefix of 0x or 0X
 	;if you make heavy use of this routine, consider $ZCALL
 	;
 	s %HD=$$FUNC(%HD)
 	q
 INT	r !,"Hexidecimal: ",%HD s %HD=$$FUNC(%HD)
 	q
 FUNC(h)
-	q:$tr(h,"E","e")<0 ""
-	n c,d,dg
-	s d=0,h=$tr(h,"abcdef","ABCDEF")
-	f c=1:1:$l(h) s dg=$f("0123456789ABCDEF",$e(h,c)) q:'dg  s d=(d*16)+(dg-2)
+	n c,d,dg,l
+	s d=0,h=$tr(h,"abcdefx","ABCDEFX"),l=$l(h)
+	s:("0X"=$e(h,1,2)) h=($zextract(h,3,l)),l=l-2
+	f c=1:1:l s dg=$f("0123456789ABCDEF",$e(h,c)) q:'dg  s d=(d*16)+(dg-2)
 	q d
```

Comment:
This may be the most efficient implementation of FUNC(h), making use of compile time evaluation of operations on constants:
```
FUNC(h)
        q:$t(h,"E","e")<0 ""
        new p,z
        s h=$tr(h,"0123456789ABCDEFabcdefXx",$zch(0)_$zch(1)_$zch(2)_$zch(3)_$zch(4)_$zch(5)_$zch(6)_$zch(7)_$zch(8)_$zch(9)_$zch(10)_$zch(11)_$zch(12)_$zch(13)_$zch(14)_$zch(15)_$zch(10)_$zch(11)_$zch(12)_$zch(13)_$zch(14)_$zch(15))
        s z=0,p=1/16 f i=$l(h):-1:1 s p=p*16,z=p*$a($e(h,i))+z
        q z
```